Assets/SEE/DataModel/DG/Attributable.cs

```diff
@@ -550,6 +550,7 @@ public override int GetHashCode()
             // we are using only those two attribute kinds to avoid unnecessary
             // computation in the hope that they suffice; nodes and edges should
             // have some attributes of this kind sufficiently different to others
+            // TODO (@koschke): If IntAttributes or StringAttributes are ever modified, won't this cause problems?
```

Comment:
Yes, that would cause problems.

Our redefined version of `Equals` is used to compare Attributables based on their content rather than any sort of id (introduced in GraphElement deriving from Attributable). It is used when we compare graphs. That was the intention al least. Yet, it turns out that we redefine `Equals` yet again in `GraphElement`. The other class deriving from it is `Graph`, which again has its own redefinition of `Equals`. So `Attributable.Equals` may actually never be used and could be removed along with `GetHashCode `altogether.

Please double-check and if that is true, remove `Attributable.Equals` and `Attributable.GetHashCode`.

Reply:
That is indeed true. I have removed `Equals` and `GetHashCode` in `Attributable`.